lib/tasks/detect-permanently-out-of-sync.js

```diff
@@ -0,0 +1,301 @@
+    const planDatabases = conf.arango.Plan.Databases;
+    planCollections = conf.arango.Plan.Collections;
+    const currentCollections = conf.arango.Current.Collections;
+    const health = conf.arango.Supervision.Health;
+    var planShards = {};
+    var curShards = {};
+    var shardToCol = {};
+
+    _.each(planCollections, function(database, dbname) {
+      _.each(database, function(collection, cname) {
+        _.each(collection.shards, function(shard, shname) {
+          if (!planShards.hasOwnProperty(dbname)) {
+            planShards[dbname] = {};
+          }
+          shardToCol[shname] = { name : collection.name, cid : cname};
+          planShards[dbname][shname] = shard;
+        });
+      });
+    });
+
+    _.each(currentCollections, function(database, dbname) {
+      _.each(database, function(collection, cname) {
+        _.each(collection, function(shard, shname) {
+          if (!curShards.hasOwnProperty(dbname)) {
+            curShards[dbname] = {};
+          }
+          curShards[dbname][shname] = shard.servers;
+        });
+      });
+    });
+
+    var counts = {};
+    var labels = {};
+    var shorts = {};
+    var shards = (i === 0) ? planShards : mismatch;
+
+    _.each(planDBServers, function (val, dbserver) {
+      ip = conf.arango.Supervision.Health[dbserver].Endpoint;
+      arango.reconnect(ip, "_system");
+      _.each(planDatabases, function(val, database) {
+
+        var localDB = (i === 0) ?
+            arango.GET("/_db/" + database + "/_api/collection").result : mismatch[database];
+
+        _.each(localDB, function (shard) {
+          if (!shard.name.startsWith("_statistics") && shards[database].hasOwnProperty(shard.name)) {
+            let c = arango.GET("/_db/" + database + "/_api/collection/" + shard.name + "/count").count;
+            if (!counts.hasOwnProperty(database)) {
+              counts[database] = {};
+              labels[database] = {};
+              shorts[database] = {};
+            }
+            if (planShards[database][shard.name].indexOf(dbserver) >= 0 &&
+                curShards[database][shard.name].indexOf(dbserver) >= 0) { // still planned and in sync
+              if (!counts[database].hasOwnProperty(shard.name)) {
+                counts[database][shard.name] = [c];
+                labels[database][shard.name] = [dbserver];
+                shorts[database][shard.name] = [conf.arango.Supervision.Health[dbserver].ShortName];
+              } else {
+                counts[database][shard.name].push(c); // keep track of counts
+                labels[database][shard.name].push(dbserver);
+                shorts[database][shard.name].push(conf.arango.Supervision.Health[dbserver].ShortName);
+              }
+            }
+          }
+        });
+      });
+    });
+
+    if (i != 0) {
+      mismatch = {};
+    }
+    _.each(counts, function(val, database) {
+      _.each(val, function(replica, shard) {
+        let nr = Object.keys(replica).length;
```

Comment:
As far as I see the `Object.keys()` call here is unnecessary, since `replica` is an array anyway, so one can call `.length` on it directly.

Reply:
Done.